src/libLookup/Lookup.cpp

```diff
@@ -1642,17 +1708,417 @@ bool Lookup::InitMining()
+        while (m_syncType != SyncType::NO_SYNC)
+        {
+            GetDSBlockFromLookupNodes(m_mediator.m_dsBlockChain.GetBlockCount(),
+                                      0);
+            GetTxBlockFromLookupNodes(m_mediator.m_txBlockChain.GetBlockCount(),
+                                      0);
+            this_thread::sleep_for(chrono::seconds(NEW_NODE_SYNC_INTERVAL));
+        }
+    };
+    DetachedFunction(1, func);
+}
+
+Peer Lookup::GetLookupPeerToRsync()
+{
+    LOG_MARKER();
+
+    std::vector<Peer> t_Peers;
+    for (auto p : m_lookupNodes)
+    {
+        if (p == m_mediator.m_selfPeer)
+        {
+            // SKIP
+        }
+        else
+        {
+            t_Peers.push_back(p);
+        }
+    }
+
+    int index = rand() % t_Peers.size();
+
+    return t_Peers[index];
+}
+
+std::vector<unsigned char> Lookup::ComposeGetLookupOfflineMessage()
+{
+    LOG_MARKER();
+
+    // getLookupOfflineMessage = [Port]
+    vector<unsigned char> getLookupOfflineMessage
+        = {MessageType::LOOKUP, LookupInstructionType::SETLOOKUPOFFLINE};
+    unsigned int curr_offset = MessageOffset::BODY;
+
+    Serializable::SetNumber<uint32_t>(getLookupOfflineMessage, curr_offset,
+                                      m_mediator.m_selfPeer.m_listenPortHost,
+                                      sizeof(uint32_t));
+    curr_offset += sizeof(uint32_t);
+
+    return getLookupOfflineMessage;
+}
+
+std::vector<unsigned char> Lookup::ComposeGetLookupOnlineMessage()
+{
+    LOG_MARKER();
+
+    // getLookupOnlineMessage = [Port]
+    vector<unsigned char> getLookupOnlineMessage
+        = {MessageType::LOOKUP, LookupInstructionType::SETLOOKUPONLINE};
+    unsigned int curr_offset = MessageOffset::BODY;
+
+    Serializable::SetNumber<uint32_t>(getLookupOnlineMessage, curr_offset,
+                                      m_mediator.m_selfPeer.m_listenPortHost,
+                                      sizeof(uint32_t));
+    curr_offset += sizeof(uint32_t);
+
+    return getLookupOnlineMessage;
+}
+
+bool Lookup::GetMyLookupOffline()
+{
+    LOG_MARKER();
+
+    // Remove selfPeerInfo from m_lookupNodes
+    auto iter = std::find(m_lookupNodes.begin(), m_lookupNodes.end(),
+                          m_mediator.m_selfPeer);
+    if (iter != m_lookupNodes.end())
+    {
+        m_lookupNodesOffline.push_back(*iter);
+        m_lookupNodes.erase(iter);
+    }
+    else
+    {
+        LOG_GENERAL(WARNING, "My Peer Info is not in m_lookupNodes");
+        return false;
+    }
+
+    SendMessageToLookupNodesSerial(ComposeGetLookupOfflineMessage());
+    return true;
+}
+
+bool Lookup::GetMyLookupOnline()
+{
+    LOG_MARKER();
+
+    auto iter = std::find(m_lookupNodesOffline.begin(),
+                          m_lookupNodesOffline.end(), m_mediator.m_selfPeer);
+    if (iter != m_lookupNodesOffline.end())
+    {
+        SendMessageToLookupNodesSerial(ComposeGetLookupOnlineMessage());
+        m_lookupNodes.push_back(*iter);
```

Comment:
same comments as prev

Reply:
Need to make sure the m_lookupNodesOffline and m_lookupNodes must have one and only one between them two